pkg/api/v20191231preview/register.go

```diff
@@ -8,25 +8,15 @@ import (
 	"github.com/Azure/ARO-RP/pkg/env"
 )
 
-const (
-	resourceProviderNamespace = "Microsoft.RedHatOpenShift"
-	resourceType              = "openShiftClusters"
-)
-
 func init() {
 	api.APIs["2019-12-31-preview"] = &api.Version{
 		OpenShiftClusterConverter: func() api.OpenShiftClusterConverter {
 			return &openShiftClusterConverter{}
 		},
 		OpenShiftClusterValidator: func(env env.Interface, resourceID string) api.OpenShiftClusterValidator {
 			return &openShiftClusterValidator{
-				sv: openShiftClusterStaticValidator{
-					location:   env.Location(),
-					resourceID: resourceID,
-				},
-				dv: openShiftClusterDynamicValidator{
-					env: env,
-				},
+				OpenShiftClusterStaticValidator:  api.NewOpenShiftClusterStaticValidator(env, resourceID, converter),
+				OpenShiftClusterDynamicValidator: api.NewOpenShiftClusterDynamicValidator(env),
```

Comment:
Here we can wrap one or both of the validators to add version-specific validation logic (will be useful in #118 for Admin API and for the future versions).